lib/run_loop/detect_aut/xcode.rb

```diff
@@ -145,6 +159,19 @@ def xcode_preferences_plist
       def pbuddy
         @pbuddy ||= RunLoop::PlistBuddy.new
       end
+
+      # @!visibility private
+      def self.detect_selected_device
+        file_name = find_user_state_files
+        selected_device = `/usr/libexec/PlistBuddy -c Print "#{file_name}"`
```

Comment:
Use `RunLoop::PlistBuddy`.

```
pbuddy = RunLoop::PlistBuddy.new
```

Expand the implementation of PlistBuddy as necessary.